xbmc/pvr/PVRManager.cpp

```diff
@@ -249,6 +252,28 @@ CPVRClientsPtr CPVRManager::Clients(void) const
   return m_addons;
 }
 
+void CPVRManager::OnSettingChanged(const CSetting *setting)
+{
+  if (setting == nullptr)
+    return;
+
+  const std::string &settingId = setting->GetId();
+  if (settingId == CSettings::SETTING_EPG_DAYSTODISPLAY)
+  {
+    m_addons->SetEPGTimeFrame(static_cast<const CSettingInt*>(setting)->GetValue());
```

Comment:
> @ksooo don't get me wrong, I don't want to be picky and I don't block this in any way. Just sharing ideas. This has been like this before, so no regression.
Hijacking the cb thread (in most cases gui thread) having it call into addons is probbly not the safest way to handle callbacks.

All fine. Thanks for the feedback. As you said, no regression here. I will address this in a nother PR. I guess there are even more places in PVR using this anti pattern. ;-)